tests/functional/syntax/modules/test_initializers.py

```diff
@@ -0,0 +1,942 @@
+"""
+tests for the uses/initializes checker
+main properties to test:
+- state usage -- if a module uses state, it must `used` or `initialized`
+- conversely, if a module does not touch state, it should not be `used`
+- global initializer check: each used module is `initialized` exactly once
+"""
+
+import pytest
+
+from vyper.compiler import compile_code
+from vyper.exceptions import (
+    BorrowException,
+    ImmutableViolation,
+    InitializerException,
+    StructureException,
+    UndeclaredDefinition,
+)
+
+
+def test_initialize_uses(make_input_bundle):
+    lib1 = """
+counter: uint256
+
+@deploy
+def __init__():
+    pass
+    """
+    lib2 = """
+import lib1
+
+uses: lib1
+
+counter: uint256
+
+@deploy
+def __init__():
+    pass
+
+@internal
+def foo():
+    lib1.counter += 1
+    """
+    main = """
+import lib2
+import lib1
```

Comment:
Would there be any issue if the imports weren't in this order?

Reply:
no, the import order is random throughout the tests (i copy-pasted mostly)

generally, the order of top-level statements in vyper doesn't matter, because there is a resolution algorithm in ModuleAnalyzer.analyze().